test/sh_do_test.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+if [ "${RESULTS_DIR}" == "" ]; then
+    echo Note: this program is normally called through the Makefile, it
+    echo is not meant to be called directly by the user.
+    exit 1
+fi
+
+# TEST_DIR should be one of compilable, fail_compilation or runnable
+export TEST_DIR=$1
+export TEST_NAME=$2
```

Comment:
There are a few more variables which I frequently need, but I think we can easily go step by step here.